platform/src/ToolsManager.js

```diff
@@ -31,8 +31,26 @@ class ToolManager {
                 
                 let toolUrl = new Object();
                 toolUrl.id = ""; // Populate when tool is fetched
-                toolUrl.url = url;
-
+                if (this.isValidUrl(url)){
```

Comment:
This does two things at once, which need different treatment. We should have a function checking if there is a use of `{{BASE-URL}}` and a separate function checking validity of the URL. Suitable error checking should be used, though note that in certain cases (editor generation) invalid URLs are initially expected and shouldn't stop proceedings.

Reply:
isValidUrl method is separated into two methods, `isUrlPlaceHolder()` checks whether the url is a placeholder or not.